src/Component/Locations.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import axios from 'axios'
+class Locations extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+            cities:[],
+            showCity: false
+        }
+    }
+
+    ghibliCities = async () =>{
+        try{
+       const { data } = await axios.get("https://ghibliapi.herokuapp.com/locations")
+    
+       this.setState({cities:data})
+        } catch (e){
+            console.log(e)
+        }
+    }
+
+    componentDidMount(){
+        this.ghibliCities()
+    }
+    handleClick = () =>{
+        if(this.state.showCity === true){
+            this.setState({showCity:false})
+        }else {
+            this.setState({showCity:true})
+        }
+    }
+    render() {
+        const { cities, showCity } =this.state
+        let city = cities.map((ghibliaCity) => {
+            return (<li key={ghibliaCity.id}>{ghibliaCity.name}</li>)
+        })
+        return (
+            <div>
+                <h2>List of Locations</h2>
+                {/* <button onClick={this.handleClick}><h6>Show Locations</h6></button> */}
+               {showCity ? (
+                <div> 
```

Comment:
Instead of grabbing the div elements it would be better to be more targeted. Something like this is specific:

`<button onClick={this.handleClick}><h6> { showCity ? "Hide" : "Show" } Locations</h6></button>`